### PR TITLE
Declare support for Python 3.10 and update links

### PR DESCRIPTION
**Pull request by hugovk**
Python 3.10 is due out in just over a couple of weeks, and is already tested by the CI. Let's add the Trove classifier.

Also in `setup.cfg` update the `url` to the new, redirected domain, and add the changelog to the project URLs; a common task when visiting PyPI is to check what changed in the most recent release.

  * Shall I also PR these link updates to the `rewrite` branch?
  * Does this need a news file?

## Thanks for contributing a pull request!

If you are contributing for the first time or provide a trivial fix don't worry too
much about the checklist - we will help you get started.

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [n/a] added/updated test(s)
- [n/a] updated/extended the documentation
- [n/a] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [ ] added news fragment in [changelog folder](../tree/master/docs/changelog)
  * fragment name: `<issue number>.<type>.rst` for example (588.bugfix.rst)
  * `<type>` is must be one of `bugfix`, `feature`, `deprecation`, `breaking`, `doc`, `misc`
  * if PR has no issue: consider creating one first or change it to the PR number after creating the PR
  * "sign" fragment with ```-- by :user:`<your username>`.```
  * please, use full sentences with correct case and punctuation, for example:
    ```rst
    Fixed an issue with non-ascii contents in doctest text files -- by :user:`superuser`.
    ```
  * also see [examples](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

**Review by gaborbernat — APPROVED** [review]

**Comment by asottile** [comment]
setup-cfg-fmt will do this automatically once 3.10 final is available (as you can see it undid your change) -- the intent is that you *don't need* these types of PRs (which I think are noisy!)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -46,6 +46,8 @@ repos:
         args:
           - --min-py3-version
           - "3.4"
+	  - max-py-version
+	  - "3.10"
```
Comment (jugmac00) [inline .pre-commit-config.yaml:50]: The indentation looks off - but I am not sure.

Comment (hugovk) [reply .pre-commit-config.yaml]: Oops, my diff tool added tabs! 😱

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**